pexpect/pxssh.py

```diff
@@ -261,6 +262,14 @@ def login (self, server, username, password='', terminal_type='ansi',
         Set ``password_regex`` if there is a MOTD message with `password` in it.
         Changing this is like playing in traffic, don't (p)expect it to match straight
         away.
+        
+        If you require to connect to another SSH server from the your original SSH
+        connection set ``spawn_local_ssh`` to `False` and this will use your current
+        session to do so. Setting this option to `False` and not having an active session
+        will trigger an error.
+        
+        Set ``ssh_key`` to `True` to force passing the current SSH authentication socket to the
```

Comment:
This logic seems incorrect. To use -A, don't you want to call the first time you ssh to your server, and it will then use that ssh authentication information if you then ssh to another remote server inside that session?

In which case, you would likely need -A in addition to your ssh_key identify file string, so the if statements down below don't make sense.

The right thing to do here might be to add a keyword argument to forward_auth=False by default, and add a -A if its set to True.

Reply:
You would use this on the 2nd ssh session to pass the original key across to the 3rd session that you might want to start. This option is used instead of specifying a key file because it assumes that you have keys loaded in the ssh key agent.

If you know the key you want to use on the remote system, you'd either load it into the current agent or pass the current agent further down the line.